### ISSUE TITLE
Incorrect dates passed to daygridMonth header

### ISSUE THREAD

**Issue by l3szcz** [fullcalendar/fullcalendar#5854]
### Reduced Test Case

[REDUCED TEST CASE](https://codesandbox.io/s/daygridmonth-header-problem-sp7rn).

### Bug Description

**Describe how to recreate the bug.**

Use props `dayHeaderFormat` or `dayHeaderContent` to change header display on `daygridMonth`. In both cases the date passed to those functions is incorrect - points to Jan 1970.

**What do you expect to happen?**

Correct date should be passed to formatting/rendering functions just like in the other views.

**What happens instead?**

Incorrect date is passed - it's always Jan 1970.

**Comment by acerix** [comment]
Thanks, confirmed. You demo doesn't run for me but I get the same in the basic debug template:

https://codepen.io/acerix/pen/MWyLrdm?editors=0011

**Comment by arshaw** [comment]
As hard as it is to believe, this behavior is intentional. The headers above the dates in month view do not belong to any of the individual days below them. They are "timeless" in a way. They do not rerender when the month changes either. To make clear the fact that they are not linked to a current-month date, I made their dates in the render-hook in the distant past.

What are you hoping to render in the headers? Something date-specific?

**Comment by l3szcz** [comment]
I would expect it to work the same as v4. [Here is forked v4 component sandbox ](https://codesandbox.io/s/fullcalendar-v4-react-forked-i00rt?file=/src/DemoApp.jsx) with enabled `columnHeaderFormat`.

If it's intended change then you can close the issue, it just looks like a bug.

**Comment by arshaw** [comment]
Thanks @l3szcz , it is intended, but I'll make a task to update the docs

**Comment by Dominik-Kohler** [comment]
I would also expect it to behave like v4:
Incorrect(v5):
![InCorrect](https://user-images.githubusercontent.com/50373709/106724235-45b20c00-6608-11eb-89f8-a0c0302b529f.png)

Correct(v4):
![Correct](https://user-images.githubusercontent.com/50373709/106724295-58c4dc00-6608-11eb-851b-e9245a9274a2.png)

As ‘dayHeaderFormat’ currently no longer works properly.

**Comment by pdcmoreira** [comment]
When FullCalendar wants to pass just dates (ignoring time) or, like in this case, just a bit of information from a date, it passes a `Date` object with the hours set to 00. The problem is that `Date` always considers the local timezone, so in places with negative offset, it becomes the previous day.

Example when working with a UTC-3 timezone:
![image](https://user-images.githubusercontent.com/4430606/161245433-2efa31a4-03bb-4296-97ec-cb135b4ba588.png)

The solution is, parse it as UTC with moment:
```js
dayHeaderContent({ date }, h) {
  moment.utc(date).format('ddd')
}
```

**Comment by olegafx** [comment]
@arshaw but how to highlight a header in a `monthly` view for `today`?

I want to set some background color if it's today. And it works perfectly in `weekly` and `day` views. But in monthly in doesn't add `fc-day-today` class

**Comment by DiarQoroviqi** [comment]
Does anybody find a solution for that issue??

**Comment by DiarQoroviqi** [comment]
@l3szcz I know its been a long time since you reported this issue but did you get any solution because I have the same problem?

**Comment by l3szcz** [comment]
> @l3szcz I know its been a long time since you reported this issue but did you get any solution because I have the same problem?

No, at our company we determined that the functionality is not that important to look for a workaround or push FullCalendar team to reverse the change.

**Comment by DiarQoroviqi** [comment]
@acerix hi Dylan, did you you guys found the issue or a solution for this ticket?

**Comment by jmginfor** [comment]
How can we correct this? It's not clear and the bug has been detected few months ago with services pack and it still doesn't work.

**Comment by Troy-Yang** [comment]
Same issue,  we customized the grid header of month view with this hook, however, we always get `"1970-01-04T00:00:00.000Z"` for the first day, so how can we get the week label correctly based on the date?

I checked the source code and fullcalendar has the below logic. The text output is correct by the dateEnv format.
```
    let date = addDays(new Date(259200000), props.dow) // start with Sun, 04 Jan 1970 00:00:00 GMT
    let dateMeta: DateMeta = {
      dow: props.dow,
      isDisabled: false,
      isFuture: false,
      isPast: false,
      isToday: false,
      isOther: false,
    }
    let text = dateEnv.format(date, props.dayHeaderFormat)
    let renderProps: DayHeaderContentArg = { // TODO: make this public?
      date,
      ...dateMeta,
      view: viewApi,
      ...props.extraRenderProps,
      text,
    }
```

**Comment by Troy-Yang** [comment]
Just find the fullcalendar actually get the text by calling formatDate from @fullcalendar/common and it eventually equals to the below codes:
```
 const weeklabel = new Intl.DateTimeFormat(locale, { timeZone: UTC, weekday: 'short' }).format(date);
```
Therefore, I found my solution now.

**Comment by anbari1337** [comment]
> @l3szcz I know its been a long time since you reported this issue but did you get any solution because I have the same problem?

Hello did you find a solution, it's been almost two years and it's not fixed yet. I have the same problem with a project I'm working on now

**Comment by palamedes** [comment]
This issue is still showing it's ugly head.
